lib/common/model.js

```diff
@@ -315,6 +348,19 @@ function ModelFactory (
             });
         },
 
+        updateAndFindMongo: function(filter, update, options){
+            var self = this;
+            return self.updateMongo(filter, update, options)
+            .then(function(){
+                if (options && options.multi === true){
+                    return self.findMongo(filter);
+                }
+                // If filter point to multiple doc and multi !== true
+                // Update will be set on a random node.
+                return self.findOneMongo(filter);
```

Comment:
They truly have a chance to operate on the same random doc.
So are you recommended here to throw error or some thing else to prevent possible mistake?